### PR TITLE
Upload-Flow Refinement

### PR DESCRIPTION
**Pull request by guyfreund**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
- Sorted packs in Slack message.
- Updated docstrings across the private content handling.
- Improved error handling across the private content handling.
- Added support for the testing flow in the `validate_premium_packs.sh` script.

## Screenshots
Paste here any images that will help the reviewer

### CODE REVIEW

File: Tests/Marketplace/upload_packs.py [hunk 1]
```diff
@@ -467,10 +467,12 @@ def update_index_with_priced_packs(private_index_path: str, extract_destination_
...
         add_private_packs_to_index(index_folder_path, private_index_path)
-        logging.info("Finished updating index with priced packs")
-    except Exception:
-        logging.exception('Could not add private packs to the index.')
+
+    except Exception as e:
+        logging.exception(f"Could not add private packs to the index. Additional Info: {str(e)}")
+
     finally:
+        logging.info("Finished updating index with priced packs")
```
Comment (abaumgarten) [inline Tests/Marketplace/upload_packs.py:475]: What do you say about:
```suggestion
        logging.info("Finished creating index with priced packs")
```

File: Tests/Marketplace/upload_packs.py [hunk 2]
```diff
@@ -629,6 +631,10 @@ def check_if_index_is_updated(index_folder_path: str, content_repo: Any, current
             logging.info("Skipping index update check in non production/build bucket")
             return
 
+        if is_private_content_updated:
+            logging.debug("Skipping index update ad Private Content has updated.")
```
Comment (abaumgarten) [inline Tests/Marketplace/upload_packs.py:635]:
```suggestion
            logging.debug("Skipping index update as Private Content has updated.")
```

File: Tests/Marketplace/upload_packs.py [hunk 3]
```diff
@@ -439,15 +439,15 @@ def build_summary_table_md(packs_input_list: list, include_pack_status: bool = F
     return '\n'.join(table)
 
 
-def update_index_with_priced_packs(private_index_path: str, extract_destination_path: str, index_folder_path: str,
-                                   pack_names: set) -> Tuple[Union[list, list], list]:
-    """ Updates index with priced packs and returns list of priced packs data.
+def add_private_content_to_index(private_index_path: str, extract_destination_path: str, index_folder_path: str,
+                                 pack_names: set) -> Tuple[Union[list, list], list]:
+    """ Adds a list of priced packs data-structures to the public index.json file.
```
Comment (abaumgarten) [inline Tests/Marketplace/upload_packs.py:444]:
```suggestion
    """ Adds a list of priced packs data-structures to the public index.json file. This step should not be skipped even if there are no new or updated private packs.
```

**Comment by abaumgarten** [comment]
@guyfreund maybe we should add to this PR the following task: https://github.com/demisto/etc/issues/34007
What do you say? I can add it

File: Tests/Marketplace/upload_packs.py (hunk 1 above)
Comment (guyfreund) [reply Tests/Marketplace/upload_packs.py:475]: > @guyfreund maybe we should add to this PR the following task: [demisto/etc#34007](https://github.com/demisto/etc/issues/34007)
> What do you say? I can add it

Let's leave it here

Comment (guyfreund) [reply Tests/Marketplace/upload_packs.py:475]: Index is not being created here

**Comment by guyfreund** [comment]
> @guyfreund maybe we should add to this PR the following task: [demisto/etc#34007](https://github.com/demisto/etc/issues/34007)
> What do you say? I can add it

No need, we can merge both PRs

**Review by abaumgarten — APPROVED** [review]

File: Tests/scripts/slack_notifier.py [hunk 4]
```diff
@@ -147,20 +147,19 @@ def get_attachments_for_bucket_upload_flow(build_url, job_name, packs_results_fi
...
-                "value": "\n".join([pack_name for pack_name in {*successful_packs}]),
+                "value": "\n".join(sorted([pack_name for pack_name in {*successful_packs}])),
                 "short": False
             }]
         if failed_packs:
             steps_fields += [{
                 "title": "Failed Packs:",
-                "value": "\n".join([f"{pack_name}: {PackStatus[pack_data.get(BucketUploadFlow.STATUS)].value}"
-                                    for pack_name, pack_data in failed_packs.items()]),
+                "value": "\n".join(sorted([pack_name for pack_name in {*failed_packs}])),
```
Comment (ShahafBenYakir) [inline Tests/scripts/slack_notifier.py:156]: Why we have decided to remove the status/reason here?

Comment (guyfreund) [reply Tests/scripts/slack_notifier.py:156]: It just looks bad and not so informative, I think it's better to have the slack msgs with the pack names only:
image

**Review by Bargenish — APPROVED** [review]

File: Tests/scripts/slack_notifier.py (hunk 4 above)
Comment (ShahafBenYakir) [reply Tests/scripts/slack_notifier.py:156]: So why do we save it if we don't use it?

**Comment by guyfreund** [comment]
Successful run: https://app.circleci.com/pipelines/github/demisto/content/70905/workflows/b0e440f6-a95b-494a-8c1c-1b0eba6d5cf5
Slack msg:
image

**Pull request merged**